contracts/truefi2/LineOfCreditAgency.sol

```diff
@@ -236,24 +236,30 @@ contract LineOfCreditAgency is UpgradeableClaimable, ILineOfCreditAgency {
      * @param pool Pool to update credit score for
      * @param borrower Borrower to update credit score for
      */
-    function updateCreditScore(ITrueFiPool2 pool, address borrower) external {
-        (uint8 oldScore, uint8 newScore) = _updateCreditScore(pool, borrower);
-        if (oldScore == newScore) {
+    function updateCreditScore(ITrueFiPool2 pool, address borrower) public {
+        (uint8 oldEffectiveScore, uint8 newEffectiveScore) = _updateCreditScore(pool, borrower, borrowed[pool][borrower]);
+        if (oldEffectiveScore == newEffectiveScore) {
             return;
         }
 
-        _rebucket(pool, borrower, oldScore, newScore, borrowed[pool][borrower]);
+        _rebucket(pool, borrower, oldEffectiveScore, newEffectiveScore, borrowed[pool][borrower]);
     }
 
     /**
      * @dev Internal function to update `borrower` credit score for `pool` using credit oracle
-     * @return Tuple containing (oldScore, newScore)
+     * @return Tuple containing (oldEffectiveScore, newEffectiveScore)
      */
-    function _updateCreditScore(ITrueFiPool2 pool, address borrower) internal returns (uint8, uint8) {
-        uint8 oldScore = creditScore[pool][borrower];
-        uint8 newScore = creditOracle.score(borrower);
-        creditScore[pool][borrower] = newScore;
-        return (oldScore, newScore);
+    function _updateCreditScore(
+        ITrueFiPool2 pool,
+        address borrower,
+        uint256 borrowedAmount
+    ) internal returns (uint8, uint8) {
+        uint8 oldEffectiveScore = creditScore[pool][borrower];
+        uint8 newEffectiveScore = creditOracle.score(borrower);
```

Comment:
If we want to be pedantic, this is not effective score but just a score, but I'll leave it up to you if it's worth changing